clients/src/main/java/org/apache/kafka/common/record/RecordBatch.java

```diff
@@ -55,6 +55,12 @@
     int NO_SEQUENCE = -1;
 
     /**
+     * For offset commit messages, we need to skip the sequence checks since sequence numbers are not
+     * generated on the producer for these messages.
+     */
+    int SKIP_SEQUENCE_CHECK = -3;
```

Comment:
I agree. I changed the code to have the `Log` exempt writes to the offsets topic from sequence number validation.